bfvmm/src/hve/arch/intel_x64/vmexit/interrupt_window.cpp

```diff
@@ -41,8 +41,16 @@ interrupt_window_handler::interrupt_window_handler(
 void
 interrupt_window_handler::queue_external_interrupt(uint64_t vector)
 {
-    if (this->is_open() && m_interrupt_queue.empty()) {
-        this->inject_external_interrupt(vector);
+    if (this->is_open()) {
+
+        if (m_interrupt_queue.empty()) {
+            this->inject_external_interrupt(vector);
+            return;
+        }
+
+        this->inject_external_interrupt(m_interrupt_queue.pop());
```

Comment:
What happens if you inject more than one interrupt? Would this cause the previously injected interrupt to be lost?

Also.... do we need to guard this queue?

Reply:
Only one will be injected. The issue with what was here before is that injection was happening only.  if the queue was empty. But it's possible to have interrupts arrive while the window is closed.

The queue is local to each vcpu so no. If we had interrupts enabled in the vmm then yes

Reply:
So that seems like an issue. If you queue up more than one and the vector is not the same, this code will drop the vector. Any ideas on how to fix this?

I also agree that the previous code was not right.